### ISSUE TITLE
Spaceship landing collision detection is overzealous when checking core seams

### ISSUE THREAD

**Issue by BurninSun** [EarendelDevelopers/factorio-mods#223]
This ship is unable to land even though visually it should easily be able to. Assuming its because the collision check just squares off the whole spaceship to do the check.

![image](https://user-images.githubusercontent.com/12116957/182015335-f4d9773b-fc75-45cc-9103-ccac93630f26.png)

**Comment by InappropriatePenguin** [comment]
I don't think this is fixable. We have to use a bounding box for the `find_entities_filtered` calls that we make to look for special entities that we don't want you to land on.

Breaking down a ship into multiple areas isn't an option either. Aside from the added complexity, we'd have to repeat `find_entities_filtered` calls for each area, which would be quite bad for performance.

**Comment by InappropriatePenguin** [comment]
I took another look, and it seems like we're _already_ doing the inefficient thing I was describing for all other _non-special_ entities. I'm not sure why both approaches are used for different searches. That does at least make this feasible to add. I'll just have to do some profiling to be sure.
